scipy/stats/_multivariate.py

```diff
@@ -5057,6 +5059,175 @@ def rvs(self, size=1, random_state=None):
                               size=size,
                               random_state=random_state)
 
+class dirichlet_multinomial_gen(multi_rv_generic):
+    r"""A Dirichlet multinomial random variable.
+
+    Methods
+    -------
+    logpmf(alpha, x, n):
+        Log of the probability mass function.
+    pmf(alpha, x, n):
+        Probability mass function.
+    mean(alpha, n):
+        Mean of the Dirichlet multinomial distribution.
+    var(alpha, n):
+        Variance of the Dirichlet multinomial distribution.
+    cov(alpha, n):
+        The covariance of the Dirichlet multinomial distribution.
+
+    Parameters
+    ----------
+    %(_dirichlet_doc_default_callparams)s
+    %(_doc_random_state)s
+
+    References
+    ----------
+    .. [1] Dirichlet-multinomial distribution,
+     https://www.wikipedia.org/wiki/Dirichlet-multinomial_distribution
+     """
+    def __init__(self, seed=None):
+        super().__init__(seed)
+        self.__doc__ = doccer.docformat(self.__doc__, dirichlet_docdict_params)
+
+    def __call__(self, alpha, seed=None):
+        return dirichlet_frozen(alpha, seed=seed)
+
+    def logpmf(self, alpha, x, n):
+        """
+        The log of the probability mass function.
+
+        Parameters
+        ----------
+        alpha: ndarray
+            The concentration parameters. The number of
+            entries determines the dimensionality of the distribution.
+        n: int
+            Number of trials.
+
+        Returns
+        -------
+        out: scalar
+            Log of the probability mass function.
+        """
+
+        if alpha.shape != x.shape:
+            raise ValueError("`x` and `alpha` must have the same shape.")
+
+        #Reduces line length
+        A = alpha
+        B = np.sum(A, A.ndim - 1)
+        C = np.add(x, 1)
+
+        out = loggamma(B) + loggamma(n + 1) - loggamma(n + B)
+        out += np.sum(loggamma(x + A) - (loggamma(A) + loggamma(C)), A.ndim - 1)
+
+        return out
+
+    def pmf(self, alpha, x, n):
+        """
+        Probability mass function for a Dirichlet multinomial distribution.
+        .. note::
+        The results for this function can be inaccurate when n,
```

Comment:
They slightly deviate, such as in test_logpmf. It wouldn't change results significantly, but if you needed extremely precise results, it could matter. That being said, I have a hard time imagining a case where that level of precision would be needed. I think it'd be better to err on the side of caution and leave it, but it's up to you.

Reply:
IIUC there are cases in which exponentiating `logpmf` could be _substantially_ more accurate, but we are not aware of cases in which it can be substantially less accurate (only slightly less accurate). If that is correct, it seems preferable to implement `pdf` by exponentiating `logpmf`. This pretty common.

Reply:
Done.